src/Foundation/Account/code/Services/Authentication/AuthenticationService.cs

```diff
@@ -14,24 +14,71 @@
 
 namespace Wooli.Foundation.Account.Services.Authentication
 {
-    using System.Diagnostics.CodeAnalysis;
+    using System.Linq;
+
+    using Base.Models;
+    using Base.Services.Pipeline;
 
     using DependencyInjection;
 
-    using Sitecore.Security.Authentication;
+    using Infrastructure.Pipelines.Login;
+    using Infrastructure.Pipelines.Logout;
+
+    using Models.Authentication;
+
+    using Sitecore.Diagnostics;
+    using Sitecore.Pipelines;
 
-    [ExcludeFromCodeCoverage]
     [Service(typeof(IAuthenticationService), Lifetime = Lifetime.Transient)]
     public class AuthenticationService : IAuthenticationService
     {
-        public void Logout()
+        private readonly IPipelineService pipelineService;
+
+        public AuthenticationService(IPipelineService pipelineService)
+        {
+            Assert.ArgumentNotNull(pipelineService, nameof(pipelineService));
+
+            this.pipelineService = pipelineService;
+        }
+
+        public Result<LoginResult> Login(string email, string password)
         {
-            AuthenticationManager.Logout();
+            var args = new LoginPipelineArgs()
+            {
+                Email = email,
+                Password = password
+            };
+
+            this.pipelineService.RunPipeline(Constants.Pipelines.Login, args);
+
+            var errorMessages = args.GetMessages(PipelineMessageFilter.Errors);
+            return !args.Aborted && errorMessages.Length > 0
```

Comment:
Do we actually have such situation when our pipeline is not aborted but we have error messages?
And do we actually need to have LoginResult now, when we can identify success or failure with `Result.Success` property?

Reply:
We have discussion with Daria and rework this logic. Later we will update FE part and this information will be needed